src/gantt.ts

```diff
@@ -789,30 +794,53 @@ module powerbi.extensibility.visual {
 
         /**
          * Calculate days off
+         * @param daysOffDataForAddition Temporary days off data for addition new one
+         * @param firstDayOfWeek First day of working week. From settings
+         * @param date Date for verifying
+         * @param extraCondition Extra condition for handle special case for last date
+         */
+        private static addNextDaysOff(
+            daysOffDataForAddition: DaysOffDataForAddition,
+            firstDayOfWeek: number,
+            date: Date,
+            extraCondition: boolean = false): DaysOffDataForAddition {
+            daysOffDataForAddition.amountOfLastDaysOff = 1;
+            for (let i = DaysInAWeekend; i > 0; i--) {
+                let dateForCheck: Date = new Date(date.getTime() + (i * MillisecondsInADay));
+                if  (dateForCheck.getDay() === +firstDayOfWeek &&
+                    (!extraCondition || (extraCondition && !/00\:00\:00/g.test(dateForCheck.toTimeString())))) {
```

Comment:
I suppose that it'd be better to move RegExp into a static variable.

Reply:
I am not sure